compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaArrayTypeSymbol.java

```diff
@@ -49,11 +53,13 @@ public TypeSymbol memberTypeDescriptor() {
 
     @Override
     public String signature() {
-        return memberTypeDescriptor().signature() + "[]";
+        // If the array is of fixed size, the signature should reflect that.
+        String sizeStr = size().map(Objects::toString).orElse("");
+        return memberTypeDescriptor().signature() + "[" + sizeStr + "]";
```

Comment:
@rasika does this particular change affects the code actions?

Reply:
@nadeeshaan seems like these are valid by the spec https://ballerina.io/spec/lang/2020R1/#section_5.3.1.1. 

```ballerina
int[1] fooResult = foo();
function foo() returns int[1]{
    
}
```